lib/internal/http2/compat.js

```diff
@@ -574,10 +574,16 @@ class Http2ServerResponse extends Stream {
     if (headers === undefined && typeof statusMessage === 'object')
       headers = statusMessage;
 
-    if (typeof headers === 'object') {
+    var i;
+    if (Array.isArray(headers)) {
+      for (i = 0; i < headers.length; i++) {
```

Comment:
The JS linter did not like that. 🤷🏼‍♂️

Reply:
Did you delete `var i` above it?

Reply:
IIRC: Yes, we don't use `let` in loops in the `lib` directory because there was a hefty performance hit for a while. While the performance penalty is now small (or perhaps even none, I'm not sure), we continue to forbid `let` in `for` loop declarations because the performance penalty still exists in the version of V8 that ships with Node.js 6.x and changes in `master` can work their way back to 6.x. Once 6.x. goes EOL in April 2019, we can remove the lint rule and go to town with `let` in `for` loop declarations if we wish.

That said, if someone wanted to make the case that 6.x is maintenance mode right now and we shouldn't be constrained by it, that would be fine by me. On the other hand, it's really not a big hindrance to not be able to use `let` in loops, especially if it's just for a few more months.